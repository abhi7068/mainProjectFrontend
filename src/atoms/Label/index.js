import React from 'react';
import { FormLabel } from '@material-ui/core';
import PropTypes from 'prop-types';
const index = (props) => {
  return (
    <div>
      <FormLabel data-testid="label" component="legend">
        {props.name}
      </FormLabel>
    </div>
  );
};
index.propTypes = {
  name: PropTypes.string,
};
export default index;
