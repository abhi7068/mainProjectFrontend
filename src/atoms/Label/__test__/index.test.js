import React from 'react';

import Label from '../index';
import { render } from '@testing-library/react';

it('it renders', () => {
  render(<Label />);
});
it('it renders with correct text', () => {
  const { getByTestId } = render(<Label name="Submit" />);
  expect(getByTestId('label')).toHaveTextContent('Submit');
});
it('should take a snapshot', () => {
  const { asFragment } = render(<Label />);
  expect(asFragment(render(<Label />))).toMatchSnapshot();
});
