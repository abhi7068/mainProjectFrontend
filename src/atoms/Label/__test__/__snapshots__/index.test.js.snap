// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <legend
      class="MuiFormLabel-root"
      data-testid="label"
    />
  </div>
</DocumentFragment>
`;
